site/app/controllers/admin/GradeableController.php

```diff
@@ -74,4 +101,76 @@ public function process_config_upload() {
         $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
             'action' => 'upload_config')));
     }
+
+    public function rename_config(){
+        $config_file_path = $_POST['curr_config_name'] ?? null;
+        if($config_file_path == null){
+            $this->core->addErrorMessage("Unable to find file");
+
+        } else {
+            $new_name = $_POST['new_config_name'] ?? null;
+            $new_dir = FileUtils::joinPaths(dirname($config_file_path, 1), $new_name);
+            if(rename($config_file_path, $new_dir)){
+                $this->core->addSuccessMessage("Successfully renamed file");
+            } else {
+                $this->core->addErrorMessage("Directory already exist, please choose another name.");
+            }
+        }
+        $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
+            'action' => 'upload_config')));
+    }
+
+    public function delete_config() {
```

Comment:
This got marked as outdated, so I'll put it here again: This method should assert that the path is in the `/var/local/submitty/{semester}/{course}/config` folder and that the provided path does not escape out of that folder.

Reply:
This was fixed, I can't delete arbitrary folders anymore